packages/ui/src/memberships/modals/EmailSubscriptionModal/EmailSubscriptionModal.tsx

```diff
@@ -13,18 +13,29 @@ import { EmailSubscriptionMachine } from './machine'
 import { EmailSubscriptionForm } from './types'
 
 export const EmailSubscriptionModal = () => {
-  // const { api } = useApi()
   const {
     hideModal,
     modalData: { member },
   } = useModal<EmailSubscriptionModalCall>()
 
+  const { wallet } = useMyAccounts()
   const [state, send] = useMachine(EmailSubscriptionMachine)
 
+  const signEmail = async () => {
+    const timestamp = Date.now()
+    const signature = await wallet?.signer.signPayload({
+      address: member.controllerAccount,
+      data: `${member.id}:${timestamp}`,
+    })
+    if (signature) {
+      send('SIGNED', { signature: signature.signature, timestamp })
+    }
+    return signature
```

Comment:
There is no need to return the signature. However if the `signature` is empty it's probably that the user cancelled the process, thus state the machine should be updated to the cancel state